### PR TITLE
Arrow crafting rebalance

### PR DESCRIPTION
**Pull request by slugyslug**
#### Summary
Balance "Removes survival as a requirement for arrow crafting, makes other small adjustments to skill requirements"

#### Purpose of change
The existing arrow crafting tree has a lot of oddities with skill requirements.

1. Almost all arrow recipes include at least one level of the survival skill, and I can't think of good reason this should be the case. Although archery is associated with survivalism, nothing about fletching actually involves natural materials except wood selection, which is not simulated in any other kind of wood crafting. Currently a master fabricator/archer can't even craft a crude arrow- which is literally just a carved stick. And having to babysit 3 skills instead of 2 increases the mental load further on a system with very low immediate reward.

2. The balancing and progression of skill requirements is all out of wack, both realistically and from a game design perspective. For example, simple wooden arrows require fab 3/sur 3/arc 1, while simple wooden small game arrows- which are the same thing but blunter- require only fab 3/sur 1. This results in a very lopsided progression of arrow power, which is hugely detrimental to DIY archery as arrow types can make or break your damage.

#### Describe the solution
I have made 3 changes:

1. I have removed all survival skill requirements from arrow recipes and from the survival bow.

2. I have nudged other skill requirements up or down in order to create a more sensible progression. Specifically:
-At 2 fab, the survival bow and crude wooden arrow unlock
-At 2 fab & 1 arc, the simple wooden arrow and simple wooden small game arrow unlock
-At 3 fab & 2 arc, the makeshift wooden arrow unlocks
-At 5 fab & 3 arc, the "real" arrows unlock: the wooden broadhead, bodkin, small game and rubber arrows

3. I have removed the makeshift metal arrow. Its name is misleading; its not an arrow with a makeshift metal shaft, its an arrow with a makeshift metal arrowhead, which is already represented in the makeshift wooden arrow. The recipes for them were practically identical- the only differences were the makeshift metal required filing and had worse stats, and the makeshift wooden arrow included other possible materials like flint. I also renamed the makeshift wooden arrow to "makeshift wooden broadhead arrow", to better communicate what part of the arrow was actually makeshift.

#### Describe alternatives you've considered
Standardizing the survival requirements could also help eliminate some wonkyness, but I erred on the side of simplicity, since burdening the archery system with three necessary skills seems excessive.

Regarding the makeshift metal arrow, there really seemed to be no reason for both makeshift arrows to exist as they both represented the same thing: a wooden arrow with an arrowhead that was not forged. I chose to keep the makeshift wooden as it is the only arrow with a possibly stone arrowhead- but it would also be possible to split the makeshift recipes into makeshift metal and stone/flint.

#### Testing
Made the changes and loaded up a new character in a new world. Everything loaded correctly and the recipes all displayed right, and crafted correctly as well.

#### Additional context
This is the second set of changes to DIY archery I am doing to make it more accessible and sensible (the first being #71348). The whole system is very cool, but because DIY archery is not very strong, I personally balked at learning its weird idiosyncrasies when I tried for the first time. Hopefully my effort will help others make the most out of a really cool system.

### CODE REVIEW

File: data/json/items/ranged/archery.json [hunk 1]
```diff
@@ -120,21 +120,6 @@
...
-    "color": "green",
-    "description": "A simple fletched wooden arrow shaft with a sharpened makeshift tip.",
-    "copy-from": "arrow_field_point_fletched",
-    "price": 0,
-    "price_postapoc": 20,
-    "damage": { "damage_type": "stab", "armor_penetration": 1 },
-    "show_stats": true,
-    "effects": [ "RECOVER_20" ],
-    "melee_damage": { "bash": 2 }
-  },
```
Comment (Karol1223) [inline data/json/items/ranged/archery.json:137]: You have not handled any obsoletion or migration for this.

**Review by Karol1223 — CHANGES_REQUESTED** [review]

File: data/json/items/ranged/archery.json (hunk 1 above)
Comment (slugyslug) [reply data/json/items/ranged/archery.json:137]: Sorry, I'm new to this. Is there a guide for that? What do I need to do for that?

Comment (TheShadowFerret) [reply data/json/items/ranged/archery.json:137]: 1.
    A. Shove the item's definition(what this comment is under) into `json/obsoletion/obsolete_items.json`
or
    B. Migrate them to makeshift wooden arrows by adding this to `obsoletion_and_migration_0.I/migration_items.json`
```
{
    "id": "arrow_metal_sharpened_fletched",
    "type": "MIGRATION",
    "replace": "arrow_wooden_heavy"
  },
```
2.  Add this to `json/obsoleteion/obsoletion_and_migration_0.I/obsolete-0.I-recipes.json` to get rid of the recipe

```
  {
    "result": "arrow_metal_sharpened_fletched",
    "type": "recipe",
    "obsolete": true
  },
```

Comment (slugyslug) [reply data/json/items/ranged/archery.json:137]: got it, thank you so much!

**Comment by slugyslug** [comment]
Have now handled obsoletion/migration

**Comment by Zireael07** [comment]
Why remove survival bow? it was a good entry level weapon

**Comment by slugyslug** [comment]
The survival bow has not been removed, only it's survival skill requirement. Edited the summary for clarity.

**Comment by slugyslug** [comment]
fixed some weirdness with merge conflicts- I just moved the item obsoletion entry higher up the list

**Pull request closed without merging**